### ISSUE TITLE
UnsafeAccessorAttribute fails to access volatile field

### ISSUE THREAD

**Issue by filipnavara** [dotnet/runtime#109665]
### Description

When trying to access `volatile string` private field through `UnsafeAccessorAttribute`, `MissingFieldException` is thrown.

### Reproduction Steps

```csharp
using System;
using System.Runtime.CompilerServices;

var a = new A();
Console.WriteLine(GetSetFoo(a));

[UnsafeAccessor(UnsafeAccessorKind.Field, Name = "foo")]
extern static ref string GetSetFoo(A args);

class A { private volatile string? foo = "Bar"; }
```

### Expected behavior

The program output is `Bar`.

### Actual behavior

```
Unhandled exception. System.MissingFieldException: Field not found: 'A.foo'.
   at Program.<<Main>$>g__GetSetFoo|0_0(A args)
   at Program.<Main>$(String[] args) in D:\unsafeaccessor\Program.cs:line 5
```

### Configuration

This seems to be broken both in .NET 8 and .NET 9 with CoreCLR runtime. Tested on Windows and macOS.

### Other information

The question of modreqs/modopts came up in the original feature proposal and this was the answer (https://github.com/dotnet/runtime/issues/81741#issuecomment-1509937128):

>> And how would this interact with modreqs and modopts that can't be specified in C# (e.g. the IsExternalInit modifier on init property setters / volatile fields etc)?
>
> Yeah, modreqs and modopts would be complicated. I think we will skip modreqs and modopts for the signature match.

This originally came up in https://github.com/KirillOsenkov/MSBuildStructuredLog/issues/834.

**Comment by filipnavara** [comment]
Additionally, this does work in both .NET 8 and .NET 9 with NativeAOT publishing.

**Comment by am11** [comment]
> This seems to be broken both in .NET 8 and .NET 9 with CoreCLR runtime.

I can repro it on osx-arm64 with .NET 9.0.0 and .NET 10.0.0-alpha.1.24553.1 (`dotnet run`) but not with .NET 8.0.10. Also, it doesn't repro on [sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEUCuA7AHwAEAmARgFgAoIgBgAIiyA6AJXwwEsBbGZgYQjcADpwA2MKAGVJAN05gYAZwDc1Ooxbs8XXswCSOyRGEyo8xaurVZAQyj1b9ALz08MAO70AggAoAlGo0ZACcvgDiMBgyGABiEBC+tv6B6qG+2rowhgBm0Ny2XBB4zLFQtrwe0ADWACLKYFCcwkV4qVTUANoAqnhKtjkw3mCWStC+vf2Dw6PQANKceAAmpZwwYkto9AByFTAu9ABEeRCH/gC61DAIGJJ4mkj0sDmaDJHRUfGJ3o5QAOZKdrqEg+egAb3owiadlu9FkEDEhXE+yYtAA/PQTgdDgAheyHFT0AC+QA===) net8.0.

**Comment by jkotas** [comment]
cc @AaronRobinsonMSFT

**Comment by AaronRobinsonMSFT** [comment]
This is likely because of

https://github.com/dotnet/runtime/blob/35d27803b83bbd46d3cfe42aa58f46f585c0c6ae/src/coreclr/vm/prestub.cpp#L1501

It might be the case we should always ignore custom modifiers for fields.

**Comment by AaronRobinsonMSFT** [comment]
Yep, setting `IgnoreCustomModifiers` to `true` makes this work, but the generated accessor isn't going to be correct since we don't currently emit the `.volatile` prefix for the `ldsfld`/`ldfld` instructions. I can see an argument that this is "unsafe" but skipping the `.volatile` really seems wrong. Let me try and special case `volatile`.

**Comment by AaronRobinsonMSFT** [comment]
Looks like there isn't any need to update the generated getters. See https://github.com/dotnet/runtime/pull/109694.
